src/applications/personalization/profile360/vet360/selectors.js

```diff
@@ -102,3 +107,30 @@ export function selectEditedFormField(state, fieldName) {
 export function selectCurrentlyOpenEditModal(state) {
   return state.vet360.modal;
 }
+
+export function selectVet360InitializationStatus(state) {
+  let status = VET360_INITIALIZATION_STATUS.UNINITALIZED;
+
+  const  { transaction, transactionRequest } = selectVet360Transaction(state, INIT_VET360_ID);
+  const isReady = selectIsVet360AvailableForUser(state);
+  let isPending = false;
+  let isFailure = false;
+
+  if (transactionRequest) {
+    isPending = transactionRequest.isPending || (transaction && isPendingTransaction(transaction));
+    isFailure = transactionRequest.isFailed || (transaction && isFailedTransaction(transaction));
+  }
+
+  if (isReady) {
+    status = VET360_INITIALIZATION_STATUS.INITIALIZED;
+  } else if (isPending) {
+    status = VET360_INITIALIZATION_STATUS.INITIALIZING;
+  } else if (isFailure) {
+    status = VET360_INITIALIZATION_STATUS.INITIALIZATION_FAILURE;
+  }
+
+  return {
+    status,
+    transaction
+  };
+}
```

Comment:
This selector is ultimately:

- Defaults the status to `UNINITIALIZED`
- If `vet360` is in the services array, then the status is `INITIALIZED`
    - An important note here is that a transaction that has been refreshed to a success status will automatically trigger a refresh of the profile and be removed from state. We're expected the refreshed profile to contain `vet360` in the services array after the initialization transaction completes.
- If there is a transactionRequest of `isPending` or an associated pending transaction, then the status is `INITIALIZING`
- If there is a failed transactionRequest or an associated failed transaction, then the status is `INITIALIZATION_FAILURE`.